client.go

```diff
@@ -147,26 +156,89 @@ func (c *httpClient) doRequest(ctx context.Context, method, url string, params R
 	// Apply mutators
 	for _, mutate := range mutators {
 		if err := mutate(req); err != nil {
+			if c.debug && c.logger != nil {
+				c.logger.Errorf("failed to mutate request: %v", err)
+			}
 			return fmt.Errorf("failed to mutate request: %w", err)
 		}
 	}
 
+	// Log the request if in debug mode
+	if c.debug && c.logger != nil {
+		var reqBody string
+		if bodyBuffer != nil {
+			reqBody = bodyBuffer.String()
+		}
+
+		reqLog := `Sending request:
+Method: {{.Method}}
+URL: {{.URL}}
+Headers: {{.Headers}}
+Body: {{.Body}}`
+		tmpl, _ := template.New("request").Parse(reqLog)
```

Comment:
This looks great, thanks for the quick turnaround!

Since the template itself isn't dynamic, you might be able to move the template creation and parsing to a top-level var so we don't need to re-parse the template every time. This also allows us to validate template errors as soon as the package is loaded using `template.Must(template.New("...")...)`.

Reply:
Fixed!